### PR TITLE
Show reactions on Comments

### PR DESCRIPTION
**Pull request by rebornix**
This is the first step of adopting Reactions API and it only displays reactions on Comments.

image

With tomorrow's Insiders, you should be able to see reactions rendered as in above image (I pushed some more change to upstream). Adding/Deleting is not supported yet as the UX is not finalized yet.

### CODE REVIEW

File: src/github/utils.ts [hunk 1]
```diff
@@ -303,6 +326,38 @@ export function convertRESTTimelineEvents(events: any[]): Common.TimelineEvent[]
...
+	let ret = [
+		{
+			title: 'CONFUSED',
+			label: '😕'
+		}, {
+			title: 'EYES',
+			label: '👀'
+		}, {
+			title: 'HEART',
+			label: '❤'
```
Comment (RMacfarlane) [inline src/github/utils.ts]: ❤️?

File: src/github/queries.gql [hunk 2]
```diff
@@ -137,6 +137,29 @@ fragment ReviewComment on PullRequestReviewComment {
...
+				content
+				user {
+					login
+					avatarUrl
+					url
+				}
+				id
+				reactable {
+					viewerCanReact
+				}
```
[the remark this answers is not in the document]
Comment (rebornix) [reply src/github/queries.gql]: Right now based on my understanding, there are only 8 supported emojis/reactions in graphql api.

File: src/github/utils.ts (hunk 1 above)
Comment (rebornix) [reply src/github/utils.ts]: I don't know how I type in that one lol

File: src/github/queries.gql (hunk 2 above)
[the remark this answers is not in the document]
Comment (RMacfarlane) [reply src/github/queries.gql]: But aren't these on a per-user basis?

[the remark this answers is not in the document]
Comment (rebornix) [reply src/github/queries.gql]: You are right, `reactions` is not the right api to use and should be replaced completely with `reactionGroup`.

File: src/github/graphql.ts [hunk 3]
```diff
@@ -42,6 +42,27 @@ export interface IssueComment {
...
+export interface ReactionGroup {
+	content: string;
+	viewerHasReacted: boolean;
+	users: {
+		totalCount: number;
+	};
+}
+
+export interface Reaction {
+	content: string;
```
[the remark this answers is not in the document]
Comment (rebornix) [reply src/github/graphql.ts]: We don't need `Reaction` anymore.

**Comment by rebornix** [comment]
@RMacfarlane comments addressed. `reactions` is no longer used in `ReviewComment`, instead we only use `reactionGroup`, which is an aggregation of 8 possible reactions on a comment.

File: src/github/graphql.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (RMacfarlane) [reply src/github/graphql.ts]: How about on content on `ReactionGroup`?

File: src/common/comment.ts [hunk 4]
```diff
@@ -114,4 +122,5 @@ export interface Comment {
 	isDraft?: boolean;
 	inReplyToId?: number;
 	graphNodeId: string;
+	reactions?: Reaction[];
```
Comment (RMacfarlane) [inline src/common/comment.ts]: I think we should also add this for the REST API case, but can do that separately

**Pull request merged**